### PR TITLE
health check and resource limits?

### PR DESCRIPTION
**Pull request by qrpike**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kube-state-metrics/92)

### CODE REVIEW

**Comment by qrpike** [comment]
I signed it!

File: kubernetes/deployment.yml [hunk 1]
```diff
@@ -15,3 +16,14 @@ spec:
...
+          httpGet:
+            path: /healthz
+            port: 8080
+            scheme: HTTP
+          initialDelaySeconds: 8
+          timeoutSeconds: 4
+        resources:
+          limits:
+            cpu: .1
+            memory: 75M
```
Comment (brancz) [inline kubernetes/deployment.yml]: While CPU resource seems fine, as `kube-state-metrics` quite literally just lists/watches, I'm wondering how you came up this as an upper bound for memory, as this highly depends on number of objects in your cluster. I'm fine with giving a suggestion of memory limits, just wondering what kind of cluster size/object count we're talking about here.

Comment (qrpike) [inline kubernetes/deployment.yml]: I just doubled/tripled what it was using in my cluster. Which is ~16nodes with ~200pods running. We can make it 100M to be safe? But it just needs some kind of limit to be safe.

**Comment by brancz** [comment]
Other than my comment something with the CLA check still doesn't seem right. Make sure that if you signed it as an individual to use the same email as you did in your git config when you created the commit.

**Comment by qrpike** [comment]
@brancz I signed it as an org, but I registered using this github account.

File: kubernetes/deployment.yml (hunk 1 above)
Comment (brancz) [inline kubernetes/deployment.yml]: `100M` limit sounds good to me then

**Comment by qrpike** [comment]
@brancz I'm curious how much data is duplicated between this and heapster? Not apparent if this is meant to work along side, or replace/be replaced by the other?

**Comment by brancz** [comment]
Heapster and kube-state-metrics are trying to solve different things. The sole purpose of `kube-state-metrics` is converting Kubernetes Objects into the Prometheus exposition format, while staying as close as possible to the data accessible when debugging things through `kubectl`. This data can then be scraped by Prometheus or Prometheus compatible systems to explore/aggregate/alert on that data.

Heapster is slowly but surely abstracted away by the Core/Custom Metrics API. Those will eventually allow pluggable solutions, so the idea is that there may be a drop in replacement for Heapster using Prometheus in the background. Regarding their differences Heapster as far as I know and can read from the [docs](https://github.com/kubernetes/heapster/blob/master/docs/overview.md) it only collects data about nodes, pods and containers, while kube-state-metrics generally collects cluster state metrics.

So to answer your question, for the time being you'll want to run both along side each other and they're all part of the puzzle.

Regarding the CLA I'll need to figure out what's happening as I can't merge if the check doesn't pass.

File: kubernetes/deployment.yml (hunk 1 above)
Comment (andyxning) [inline kubernetes/deployment.yml]: How about we change `cpu: .1` to `cpu: 100m`. This makes it more clear. :)

Comment (andyxning) [inline kubernetes/deployment.yml]: BTW, maybe we should set resource configuration like what is done for [dashboard](https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dashboard/dashboard-controller.yaml) by setting both `limits` and `requests` to make its QoS guranteed.

Comment (brancz) [inline kubernetes/deployment.yml]: I agree with both of @andyxning's comments.

**Comment by brancz** [comment]
Have you been able to figure out what the problem with the cla is?

**Comment by andyxning** [comment]
@qrpike Could you please take some time to resolve the CLA and update and rebase the code. :) We need this change.

**Comment by andyxning** [comment]
close this pr via #238 . @qrpike Feel free  to reopen it if you have any doubts.

**Pull request closed without merging**